Polyhedron/demo/Polyhedron/Plugins/IO/STL_io_plugin.cpp

```diff
@@ -1,25 +1,44 @@
+#include "Scene_surface_mesh_item.h"
 #include "Scene_polyhedron_item.h"
 #include "Scene_polygon_soup_item.h"
 #include "Kernel_type.h"
+#include "Scene.h"
 #include "Polyhedron_type.h"
 
 #include <CGAL/Three/Polyhedron_demo_io_plugin_interface.h>
+#include <CGAL/Three/Polyhedron_demo_plugin_interface.h>
+#include <CGAL/Three/Polyhedron_demo_plugin_helper.h>
 #include <fstream>
 
 #include <CGAL/IO/Polyhedron_builder_from_STL.h>
 #include <CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h>
 
+
 #include <QColor>
+#include <QMainWindow>
 using namespace CGAL::Three;
 class Polyhedron_demo_stl_plugin :
   public QObject,
-  public Polyhedron_demo_io_plugin_interface
+  public Polyhedron_demo_io_plugin_interface,
+  public Polyhedron_demo_plugin_helper
 {
   Q_OBJECT
-  Q_INTERFACES(CGAL::Three::Polyhedron_demo_io_plugin_interface)
+  Q_INTERFACES(CGAL::Three::Polyhedron_demo_plugin_interface CGAL::Three::Polyhedron_demo_io_plugin_interface)
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.PluginInterface/1.0")
   Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.IOPluginInterface/1.0")
 
 public:
+  void init(QMainWindow* mainWindow,
+            CGAL::Three::Scene_interface* scene_interface,
+            Messages_interface*) {
+    //get the references
+    this->scene = scene_interface;
+    this->mw = mainWindow;
+  }
+  QList<QAction*> actions() const {
+    return QList<QAction*>();
+  }
+  bool applicable(QAction*) const { return false;}
```

Comment:
Why is that I/O plugin turned into a regular plugin, with no action? Just to get the access to `this->mw` and then to the property?!